trio/_core/_multierror.py

```diff
@@ -328,7 +328,7 @@ def catch(
         return MultiErrorCatcher(handler)
 
 
-if TYPE_CHECKING:
+if TYPE_CHECKING:  # noqa: SIM108
```

Comment:
I agree ternaries don't make sense for `if TYPE_CHECKING` blocks but do type checkers even support it?

Purely out of curiosity!

Reply:
I'm not completely sure off the top of my head, hence the noqa I added. I would assume not unless they are doing some very fine grained logic analysis.

Reply:
ternaries aren't actually very complex AST-wise (pretty much identical to handling `if`-statements), so I wouldn't be shocked if it did work in some of them. But I could see them intentionally not supporting them.

Reply:
Allowed to change into ternary in most recent commit as of now